lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -753,4 +727,122 @@ sub _create_download_lists {
     }
 }
 
+sub _schedule_from_yaml ($self, $args, $skip_chained_deps, @load_yaml_args) {
```

Comment:
This code is similar to `_generate_jobs` but only *similar*. Not sure whether it makes sense to unify it.